modules/ingestor/src/graph/mod.rs

```diff
@@ -28,3 +29,38 @@ impl Graph {
         Self { db }
     }
 }
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum Outcome<T> {
+    Existed(T),
```

Comment:
minor: I do not fully understand the decomposition of Existed and Added state .... I mean I guess we could infer on True/False on Added ... so we want to sometimes add over an existed ... a few words explaining the intent of Outcome probably would help.